File: src/main/java/pageobjects/BurgerMainPage.java
package pageobjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import java.util.List;

public class BurgerMainPage extends Page{
    private WebDriver driver;
    private By loginButton = By.xpath("//button[text()='Войти в аккаунт']");
    private By profile = By.xpath("//div/header/nav/a");
    private List<String> sectionNames = List.of("Булки","Соусы","Начинки");
    private List <By> sectionCaptions = List.of(By.xpath("//*[text()='Булки']"),
            By.xpath("//*[text()='Соусы']"),
            By.xpath("//*[text()='Начинки']"));
    private List <By> sections = List.of(By.xpath("//*/div/main/section[1]/div[2]/h2[1]"),
            By.xpath("//*/div/main/section[1]/div[2]/h2[2]"),
            By.xpath("//*/div/main/section[1]/div[2]/h2[3]"));
    private final int sectionsQuantity = sections.size();
    private By constructor = By.xpath("//*/div/header/nav/ul/li[1]/a");
    private By logout = By.xpath("//*[text()='Выход']");

    private By logo = By.xpath("//div/a[@href='/']");

    public By getLoginButton() {
        return loginButton;
    }
    public By getProfile() {
        return profile;
    }
    public int getSectionsQuantity() {
        return sectionsQuantity;
    }
    public void openConstructor() {
        driver.findElement(constructor).click();
        wait(By.xpath("//*[text()='Соберите бургер']"));
    }
    public String getConstructorOpenedHeader(){
       return wait(By.xpath("//*[text()='Соберите бургер']")).getText();
    }
    public void openProfile() {
        driver.findElement(profile).click();
        wait(By.xpath("//*[text()='Профиль']"));
    }
    public String getProfileOpenedHeader(){
        return wait(By.xpath("//*[text()='Профиль']")).getText();
    }
    public void openMainPage(){
        driver.findElement(logo).click();
        wait(By.xpath("//*[text()='Соберите бургер']"));
    }
    public void logoutMainPage(){
        driver.findElement(logout).click();
        wait(By.xpath("//*[text()='Вход']"));
    }
    public String logoutHeader(){
        return wait(By.xpath("//*[text()='Вход']")).getText();
    }

    public BurgerMainPage(WebDriver driver){
        super(driver);
        this.driver = driver;
    }
    public void clickLoginButton(By button){
        driver.findElement(button).click();
    }
    public void login(By button, String email, String pwd){
        clickLoginButton(button);
        wait(By.xpath("//*[text()='Вход']"));
        driver.findElement(By.xpath("//fieldset[1]/div/div/input")).sendKeys(email);
        driver.findElement(By.xpath("//fieldset[2]/div/div/input")).sendKeys(pwd);
        driver.findElement(By.xpath("//button[text()='Войти']")).click();
    }
    public boolean isLoggedIn(){
        return wait(By.xpath("//*[text()='Соберите бургер']")).getText().equals("Соберите бургер");
    }

    public int identifySection(){
        String current = driver.findElement(By.className("tab_tab_type_current__2BEPc")).getText();
        return sectionNames.indexOf(current);
    }
    public void waitForNSectionVisibility(int n){
        driver.findElement(sectionCaptions.get(n)).click();
        wait(returnNSection(n));
    }
    public By returnNSection(int n){
        return sections.get(n);
    }
}

File: src/main/java/pageobjects/BurgerMainPage.kt
package pageobjects

import org.openqa.selenium.By
import org.openqa.selenium.By.xpath
import org.openqa.selenium.WebDriver
import org.openqa.selenium.support.PageFactory


class BurgerMainPage(private val driver: WebDriver) : Page(driver) {
    companion object{
        val loginButtonAuth: By = xpath("//button[text()='Войти в аккаунт']")
        val profileButtonAuth: By = xpath("//div/header/nav/a")
    }

    val loginButton: By = xpath("//button[text()='Войти в аккаунт']")
    val profile: By = xpath("//div/header/nav/a")
    private val sectionNames = listOf("Булки", "Соусы", "Начинки")
    private val sectionCaptions = listOf(xpath("//*[text()='Булки']"),
            xpath("//*[text()='Соусы']"),
            xpath("//*[text()='Начинки']"))
    private val sections = listOf(xpath("//*/div/main/section[1]/div[2]/h2[1]"),
            xpath("//*/div/main/section[1]/div[2]/h2[2]"),
            xpath("//*/div/main/section[1]/div[2]/h2[3]"))

    val sectionsQuantity = sections.size
    private val constructor = xpath("//*/div/header/nav/ul/li[1]/a")
    private val logout = xpath("//*[text()='Выход']")
    private val logo = xpath("//div/a[@href='/']")
    init {
        PageFactory.initElements(driver,this)
    }
    fun openConstructor() {
        driver.findElement(constructor).click()
        wait(xpath("//*[text()='Соберите бургер']"))
    }

    fun getConstructorOpenedHeader(): String {
        return wait(xpath("//*[text()='Соберите бургер']")).text
    }

    fun openProfile() {
        driver.findElement(profile).click()
        wait(xpath("//*[text()='Профиль']"))
    }

    fun getProfileOpenedHeader(): String {
        return wait(xpath("//*[text()='Профиль']")).text
    }

    fun openMainPage() {
        driver.findElement(logo).click()
        wait(xpath("//*[text()='Соберите бургер']"))
    }

    fun logoutMainPage() {
        driver.findElement(logout).click()
        wait(xpath("//*[text()='Вход']"))
    }

    fun logoutHeader(): String {
        return wait(xpath("//*[text()='Вход']")).text
    }

    fun clickLoginButton(button: By?) {
        driver.findElement(button).click()
    }

    fun login(button: By?, email: String?, pwd: String?) {
        clickLoginButton(button)
        wait(xpath("//*[text()='Вход']"))
        driver.findElement(xpath("//fieldset[1]/div/div/input")).sendKeys(email)
        driver.findElement(xpath("//fieldset[2]/div/div/input")).sendKeys(pwd)
        driver.findElement(xpath("//button[text()='Войти']")).click()
    }

    fun isLoggedIn(): Boolean {
        return wait(xpath("//*[text()='Соберите бургер']")).text == "Соберите бургер"
    }

    fun identifySection(): Int {
        val current = driver.findElement(By.className("tab_tab_type_current__2BEPc")).text
        return sectionNames.indexOf(current)
    }

    fun waitForNSectionVisibility(n: Int) {
        driver.findElement(sectionCaptions[n]).click()
        wait(returnNSection(n))
    }

    fun returnNSection(n: Int): By {
        return sections[n]
    }


}

